test/test_admin_basic.py

```diff
@@ -147,6 +147,45 @@ def test_disable_enable_user(self, user):
     def test_user_roles(self, user):
         assert len(user.roles()) == 0
 
+    def test_set_password_pre11_no_current_password_required(self):
+
+        mock_client = Mock()
+
+        # Mock version response (pre-11)
+        mock_client.get.return_value.json.return_value = {
+            "dbms.version": {"value": "10.1.0"}
+        }
+
+        user = admin.User("testuser", mock_client)
+        user.set_password("newpass")
+
+        mock_client.put.assert_called_once_with(
+            "/admin/users/testuser/pwd", json={"password": "newpass"}
+        )
+
+    def test_set_password_v11_requires_current_password(self):
```

Comment:
Does it make sense to do this test against the real endpoint instead of the mock? We should always be testing pystardog against 11+ right?

Reply:
We are doing both. I modified the existing test which actually hits Stardog. Then I added 2 additional tests (pre v11 and v11+). 

https://github.com/stardog-union/pystardog/pull/184/files#diff-9f12cf0e19923318dbf8916739057e59eb68033130318227be134b0a202a80f6L122